source/particle/property/crystal_preferred_orientation.cc

```diff
@@ -705,10 +790,69 @@ namespace aspect
               {
                 const Tensor<1,3> slip_normal_global = rotation_matrix_transposed*slip_normal_reference[slip_system_i];
                 const Tensor<1,3> slip_direction_global = rotation_matrix_transposed*slip_direction_reference[slip_system_i];
-                const Tensor<2,3> slip_cross_product = outer_product(slip_direction_global,slip_normal_global);
+                const Tensor<2,3> slip_cross_product = outer_product(slip_direction_global,slip_normal_global); //<Tian?> is outer_product really doing cross product?
                 bigI[slip_system_i] = scalar_product(slip_cross_product,strain_rate_nondimensional);
               }
 
+            //   //quick test for CPX slip-system Tian begin
```

Comment:
Again delete all commented code. If this is a test for your code, then make it into a test, but dont keep it here.